go/vt/vtgate/evalengine/comparisons.go

```diff
@@ -261,39 +313,21 @@ func evalCompare(lVal, rVal EvalResult) (comp int, err error) {
 	}
 }
 
-func evalCompareTuplesNullSafe(lVal EvalResult, rVal EvalResult) (int, error) {
+func evalCompareTuplesNullSafe(lVal EvalResult, rVal EvalResult) (bool, error) {
 	if len(*lVal.tuple) != len(*rVal.tuple) {
-		return 0, vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.OperandColumns, "Operand should contain %d column(s)", len(*lVal.tuple))
+		panic("did not typecheck cardinality")
```

Comment:
should we not just return an error here?

Reply:
This is on purpose: to match MySQL's behavior, we need to perform the cardinality checks early. If by the time we've reached this function the cardinalities do not match, then that's a bug, not an error.